py-pysal: update with new "family"-ports

nilason opened the pull request:
#### Description

<!-- Note: it is best to make pull requests from a branch rather than from master -->

The PySAL (`py-pysal`) package has been refactored, it is now a wrapper package for a number of separate modules.

This PR updates py-pysal and adds the separate modules as new ports.


###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS 12.6.6 21G646 arm64
Xcode 14.2 14C18

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

nilason commented:
Depends on #19719

nilason commented:
Oh dear, finally the CI builds succeeded! Please review!

reneeotten commented on python/py-access/Portfile:
```diff
@@ -0,0 +1,42 @@
+                    port:py${python.version}-numpy \
+                    port:py${python.version}-pandas
+
+    livecheck.type  none
```
this is the default set by the `python` PG and can thus be removed

reneeotten commented on python/py-access/Portfile:
```diff
@@ -0,0 +1,42 @@
+PortGroup           github 1.0
+PortGroup           python 1.0
+
+github.setup        pysal access 1.1.9 v
```
we typically prefer to fetch Python ports from PyPI - is there a reason as to why you not do that here?

reneeotten commented on python/py-momepy/Portfile:
```diff
@@ -0,0 +1,46 @@
+
+python.versions     39 310 311
+python.pep517       yes
+python.pep517_backend setuptools
```
this is the default and shouldn't be specified

reneeotten commented on python/py-pysal/Portfile:
```diff
@@ -4,28 +4,52 @@ PortSystem              1.0
 revision                0
-name                    py-pysal
+name                    py-${github.project}
+github.tarball_from     tarball
```
why not `archive`? If it needs to be `tarball` then you can omit it, as that is the current default in the `github` PG

reneeotten commented on python/py-spopt/Portfile:
```diff
@@ -0,0 +1,70 @@
+    test.cmd        py.test-${python.branch}
+    test.target
+    test.env-append \
+                    PYTHONPATH=${worksrcpath}/build/lib
```
these are the default set up by the `python` PG ; so you'll only need to retain `test.run yes`

reneeotten commented on python/py-spopt/Portfile:
```diff
@@ -0,0 +1,70 @@
+                    port:py${python.version}-setuptools_scm
+
+    depends_test-append \
+                    port:py${python.version}-pytest \
```
this is added by default when setting `test.run yes` so the `pytest` dependency can be removed here

reneeotten commented on python/py-spopt/Portfile:
```diff
@@ -0,0 +1,70 @@
+    test.env-append \
+                    PYTHONPATH=${worksrcpath}/build/lib
+
+    livecheck.type  none
```
default

reneeotten reviewed (changes requested):
thanks @nilason for all your work on these ports! I've left a few comments - they all apply to pretty much every port in this PR.

nilason commented:
> thanks @nilason for all your work on these ports! I've left a few comments - they all apply to pretty much every port in this PR.

I believe I have addressed all your points. I kept one port (py-spopt) installing from Github and therefore enabling the tests.

(Some other modules also did have tests, but they seemed unreliable.)

reneeotten commented:
thanks @nilason !

nilason commented:
@reneeotten Thank you!
